saleor/order/actions.py

```diff
@@ -151,23 +152,23 @@ def fulfillment_tracking_updated(
     get_plugins_manager().order_updated(fulfillment.order)
 
 
-def cancel_fulfillment(fulfillment: "Fulfillment", user: "User", restock: bool):
+def cancel_fulfillment(
```

Comment:
This should be a transaction. Probably we should also use `select_for_update` for the fulfillment object.